src/Core/Framework/Script/Execution/ScriptLoader.php

```diff
@@ -38,25 +36,47 @@ public function __construct(
 
     public static function getSubscribedEvents(): array
     {
-        return ['script.written' => 'invalidateCache'];
+        return [
+            'script.written' => 'invalidateCache',
+            'app.written' => 'invalidateCache',
+        ];
     }
 
     /**
      * @return Script[]
      */
     public function get(string $hook): array
     {
+        $hookScripts = [];
+
         $cacheItem = $this->cache->getItem(self::CACHE_KEY);
         if ($cacheItem->isHit() && $cacheItem->get()) {
-            return CacheCompressor::uncompress($cacheItem)[$hook] ?? [];
+            $hookScripts = CacheCompressor::uncompress($cacheItem)[$hook] ?? [];
+        } else {
+            $scripts = $this->load();
+
+            $cacheItem = CacheCompressor::compress($cacheItem, $scripts);
+            $this->cache->save($cacheItem);
+
+            $hookScripts = $scripts[$hook] ?? [];
         }
 
-        $scripts = $this->load();
+        /** @var Script[] $hookScripts */
```

Comment:
please do not use var annotations.
also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md

Reply:
if there is no other way, it is okay. I just comment that on every occurrence :grin: